src/Processors/QueryPlan/PartsSplitter.cpp

```diff
@@ -174,23 +176,20 @@ struct PartsRangesIterator
         else if (compare_result == 1)
             return false;
 
-        if (part_index == other.part_index)
+        if (event == other.event)
```

Comment:
could you pls give an example when strict weak ordering wasn't held before?
maybe there is some less radical way to fix other than allow overlapping ranges?

Reply:
Added comment in pull request. We need to have `Start event` always before `End event`. We cannot change this property only for ranges inside single part.